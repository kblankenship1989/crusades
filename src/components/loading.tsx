import React from 'react';
import {Image, View, StyleSheet} from 'react-native';

export const LoadingGIF : React.FC = () => (
    <View style={styles.container}>
        <Image
            source={require('../assets/images/loading.gif')}

            style={styles.image}
        />
    </View>
);

const styles = StyleSheet.create({
    image: {
        flex: 1,
        width: '60%',
        resizeMode: 'contain'
    },
    container: {
        flex: 1,
        alignItems: 'center',
    }
});
