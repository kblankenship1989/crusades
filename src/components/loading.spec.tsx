import React from 'react';
import {render} from '@testing-library/react-native';

import {LoadingGIF} from './loading';

describe('Given the Loading component', () => {
    it('should render correctly', () => {
        const component = (<LoadingGIF />);

        expect(render(component).toJSON()).toMatchSnapshot();
    });
});
