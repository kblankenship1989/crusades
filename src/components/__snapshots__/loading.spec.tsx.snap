// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given the Loading component should render correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
    }
  }
>
  <Image
    source={1}
    style={
      Object {
        "flex": 1,
        "resizeMode": "contain",
        "width": "60%",
      }
    }
  />
</View>
`;
